app/elements/batch-image-search/batch-image-search.html

```diff
@@ -0,0 +1,379 @@
+
+        <div class="layout horizontal">
+          <span class="name right-space">Supported file types:</span>
+          <span class="text">All Image files (.jpg, .png or .jpeg etc.)</span>
+        </div>
+
+        <paper-textarea label="Enter or Paste Image Url" value="{{_imageInput}}"" on-input="userActivity" no-label-float></paper-textarea>
+
+        <file-upload accept="image/*" id="fileUpload" droppable="true" method="POST" raised="true" target="/uploadImage" on-success="_handleUpload" multi="false">Upload Image...</file-upload>
+
+        <div class="layout horizontal">
+          <button-action class="right-space" disabled="{{boolButton}}" text="Search" click-listener="[[_createSearchListener()]]"></button-action>
+
+          <button-action id="clearAll" class="right-space" text="Clear" click-listener="[[_createClearListener()]]"></button-action>
+          <loading-spinner show="[[loading]]" type="Similar Images"></loading-spinner>
+        </div>
+        
+        <iron-ajax
+        id="xhr"
+        handle-as="json"
+        with-credentials="true"
+        loading="{{loading}}"
+        on-response="handleResponse"></iron-ajax>
+
+      </div>
+    </paper-dialog>
+
+    <paper-dialog id="resultImages" modal entry-animation="fade-in-animation" exit-animation="fade-out-animation">
+        <div class="layout vertical">
+          <div class="layout horizontal">
+            <span class="bold tall flex">Similar Images Result</span>
+            <paper-icon-button icon="clear" dialog-confirm title="Close"></paper-icon-button>
+          </div>
+          <div class="divider"></div>
+          <div class="layout vertical">
+            <span class="bold tall flex">Compared Image</span>
+              <image-thumbnail
+              source="[[compareSourceImage]]"
+              style-class="[[getBlurStyleClass(blur)]]"
+              target="_blank"
+              class="simImages"
+              text="open image">
+            </image-thumbnail>
+          </div>
+          <div class="layout">
+            <div class="divider"></div>
+            <template is="dom-repeat" items="[[images]]" rendered-item-count="{{renderedCount}}">
+              <image-thumbnail-with-link
+                link = {{displayLink(index)}}
+                source="[[item]]"
+                style-class="[[getBlurStyleClass(blur)]]"
+                target="_blank"
+                class="simImages"
+                text="open image">
+              </image-thumbnail-with-link>
+            </template>
+            <template is="dom-if" if="{{!renderedCount}}">
+              <span class="text">No Similar Images </span>
+            </template> 
+          </div>
+        </div>
+
+      </paper-dialog>
+  </template>
+
+  <script>
+  (function() {
+    'use strict';
+
+    Polymer({
+      is: 'batch-image-search',
+
+      properties: {
+        // memex Username
```

Comment:
Variable and function header comments should be in the style:
```
/**
 * Comment here.
 */
```
If you're interested, you can read more here:  https://www.polymer-project.org/1.0/docs/tools/documentation